subsys/bluetooth/mesh/access.c

```diff
@@ -2086,15 +2096,30 @@ int bt_mesh_comp_store(void)
 
 	err = bt_mesh_comp_data_get_page_0(&buf, 0);
 	if (err) {
-		LOG_ERR("Failed to read composition data: %d", err);
+		LOG_ERR("Failed to read composition data P1: %d", err);
+		return err;
+	}
+
+	err = settings_save_one("bt/mesh/cmp0", buf.data, buf.len);
+	if (err) {
+		LOG_ERR("Failed to store composition data P0: %d", err);
+	} else {
+		LOG_DBG("Stored composition data P0");
+	}
+
+	net_buf_simple_reset(&buf);
+	err = bt_mesh_comp_data_get_page_1(&buf);
```

Comment:
Shouldn't this be present only if remote provisioning feature is present?

Reply:
Im not sure if the RPR server must be present, but I think `CONFIG_BT_MESH_COMP_PAGE_1` clause should be added.